jdbc/src/main/java/org/apache/zeppelin/jdbc/JDBCInterpreter.java

```diff
@@ -0,0 +1,413 @@
+            .add(COMMON_MAX_LINE, MAX_LINE_DEFAULT,
+                "Max number of SQL result to display.").build());
+  }
+
+  public JDBCInterpreter(Properties property) {
+    super(property);
+    propertiesMap = new HashMap<>();
+    propertyKeyUnusedConnectionListMap = new HashMap<>();
+    paragraphIdStatementMap = new HashMap<>();
+    paragraphIdConnectionMap = new HashMap<>();
+  }
+  
+  public HashMap<String, Properties> getPropertiesMap() {
+    return propertiesMap;
+  }
+
+  @Override
+  public void open() {
+    for (String propertyKey : property.stringPropertyNames()) {
+      logger.debug("propertyKey: {}", propertyKey);
+      String[] keyValue = propertyKey.split("\\.", 2);
+      if (2 == keyValue.length) {
+        logger.info("key: {}, value: {}", keyValue[0], keyValue[1]);
+        Properties prefixProperties;
+        if (propertiesMap.containsKey(keyValue[0])) {
+          prefixProperties = propertiesMap.get(keyValue[0]);
+        } else {
+          prefixProperties = new Properties();
+          propertiesMap.put(keyValue[0], prefixProperties);
+        }
+        prefixProperties.put(keyValue[1], property.getProperty(propertyKey));
+      }
+    }
+
+    Set<String> removeKeySet = new HashSet<>();
+    for (String key : propertiesMap.keySet()) {
+      if (!COMMON_KEY.equals(key)) {
+        Properties properties = propertiesMap.get(key);
+        if (!properties.containsKey(DRIVER_KEY) || !properties.containsKey(URL_KEY)) {
+          logger.error("{} will be ignored. {}.{} and {}.{} is mandatory.",
+              key, DRIVER_KEY, key, key, URL_KEY);
+          removeKeySet.add(key);
+        }
+      }
+    }
+
+    for (String key : removeKeySet) {
+      propertiesMap.remove(key);
+    }
+
+    logger.debug("propertiesMap: {}", propertiesMap);
+  }
+  
+  public Connection getConnection(String propertyKey)  throws ClassNotFoundException, SQLException {
+    Connection connection = null;
+    if (propertyKeyUnusedConnectionListMap.containsKey(propertyKey)) {
+      ArrayList<Connection> connectionList = propertyKeyUnusedConnectionListMap.get(propertyKey);
+      if (0 != connectionList.size()) {
+        connection = propertyKeyUnusedConnectionListMap.get(propertyKey).remove(0);
+        if (null != connection && connection.isClosed()) {
+          connection.close();
+          connection = null;
+        }
+      }
+    }
+    if (null == connection) {
+      Properties properties = propertiesMap.get(propertyKey);
+      logger.info(properties.getProperty(DRIVER_KEY));
+      Class.forName(properties.getProperty(DRIVER_KEY));
+      String url = properties.getProperty(URL_KEY);
+      String user = properties.getProperty(USER_KEY);
+      String password = properties.getProperty(PASSWORD_KEY);
+      if (null != user && null != password) {
+        connection = DriverManager.getConnection(url, user, password);
+      } else {
+        connection = DriverManager.getConnection(url, properties);
+      }
```

Comment:
@whisperstream Hi, sorry for late reply. I agree with you and think it's my mistake and we have to fix that case that you told the above. Do you mind sharing your time to fix it?